test/cbmc/proofs/OTA_CBOR_Encode_GetStreamRequestMessage/OTA_CBOR_Encode_GetStreamRequestMessage_harness.c

```diff
@@ -0,0 +1,110 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/**
+ * @file OTA_CBOR_Encode_GetStreamRequestMessage_harness.c
+ * @brief Implements the proof harness for OTA_CBOR_Encode_GetStreamRequestMessage function.
+ */
+/* Include headers for cbor parsing. */
+#include "cbor.h"
+#include "ota_cbor_private.h"
+
+#define OTA_MAX_BLOCK_BITMAP_SIZE    128U                                                 /*!< @brief Max allowed number of bytes to track all blocks of an OTA file. Adjust block size if more range is needed. */
+#define OTA_REQUEST_MSG_MAX_SIZE     ( 3U * OTA_MAX_BLOCK_BITMAP_SIZE )                   /*!< @brief Maximum size of the message */
+
+
+/* Stub to initialize the encoder. */
+void cbor_encoder_init( CborEncoder * encoder,
+                        uint8_t * buffer,
+                        size_t size,
+                        int flags )
+{
+}
+
+/* Stub to create a cbor map for cbor stream. */
+CborError cbor_encoder_create_map( CborEncoder * encoder,
+                                   CborEncoder * mapEncoder,
+                                   size_t length )
+{
+    CborError err;
+
+    return err;
+}
+
+/* Stub to encode the text strings. */
+CborError __CPROVER_file_local_cbor_h_cbor_encode_text_stringz( CborEncoder * encoder,
+                                                                const char * string )
+{
+    CborError err;
+
+    return err;
+}
+
+/* Stub to encode integer values. */
+CborError cbor_encode_int( CborEncoder * encoder,
+                           int64_t value )
+{
+    CborError err;
+
+    return err;
+}
+
+/* Stub to append the string to a CBOR stream. */
+CborError cbor_encode_byte_string( CborEncoder * encoder,
+                                   const uint8_t * string,
+                                   size_t length )
+{
+    CborError err;
+
+    return err;
+}
+
+/* Stub to close the CBOR container. */
+CborError cbor_encoder_close_container_checked( CborEncoder * encoder,
+                                                const CborEncoder * containerEncoder )
+{
+    CborError err;
+
+    return err;
+}
+
+/* Stub to return the size of buffer. */
+size_t __CPROVER_file_local_cbor_h_cbor_encoder_get_buffer_size( const CborEncoder * encoder,
+                                                                 const uint8_t * buffer )
+{
+    size_t bufferSize;
+
+    return bufferSize;
+}
+
+void OTA_CBOR_Encode_GetStreamRequestMessage_harness()
+{
+    uint8_t * pMessageBuffer;
+    size_t messageBufferSize;
+    size_t * pEncodedMessageSize;
+    char * pClientToken;
+    int32_t fileId;
+    int32_t blockSize;
+    int32_t blockOffset;
+    uint8_t * pBlockBitmap;
+    size_t blockBitmapSize;
+    int32_t numOfBlocksRequested;
+    size_t clientTokenSize;
+
+    /* The pMessageBuffer pointer is pointing to an array of size messageBufferSize. */
+    pMessageBuffer = ( uint8_t * ) malloc( messageBufferSize * sizeof( uint8_t ) );
```

Comment:
This memory needs to be freed before the function returns. Also, please add an assume statement in case we fail to allocate to memory. Please add this to the rest of the mallocs as well.

Reply:
In this proof, we also have to check for cases where the malloc fails. I don't think we require any assume statements in this proof.

Reply:
If we don't check for the return value of malloc, does that mean it will be set to NULL in the tests some of the time?

Reply:
Yes. Sometimes the malloc can fails and when it fails it would return NULL. CBMC would consider this as a test case.